app/components/Settings/Settings.react.js

```diff
@@ -252,6 +255,27 @@ class Settings extends Component {
         }
     }
 
+    updateSelectedPanel(panelIndex, cb) {
+      this.props.updatePreview({
+          showChart: false,
+          showEditor: true,
+          size: 200
+      });
+      this.setState({
+        selectedPanel: {
+          [this.props.selectedTab]: panelIndex
+        }
+      }, typeof cb === 'function' ? cb : () => {});
+    }
+
+    openScheduler() {
+      this.setState({ scheduledQuery: this.props.preview.code }, () => {
```

Comment:
I don't understand the purpose of this code.

Also note that the value of `this.props.preview.code` isn't properly set before the connection has successfully connected to the database and retrieve the list of available tables.